### PR TITLE
Fix some pyright issues

### PR DESCRIPTION
**Pull request by david-pl**
Got it down to 14 errors (starting out with 69). Not sure if all changes make sense.

The errors that are left are mostly ones such as this:

```python
kirin-workspace/bloqade-circuit/src/bloqade/stim/emit/stim.py:35:9 - error: Method "emit_block" overrides class "EmitABC" in an incompatible manner
    Return type mismatch: base method returns type "str", override returns type "str | None"
      Type "str | None" is not assignable to type "str"
        "None" is not assignable to "str" (reportIncompatibleMethodOverride)
```

I'm not sure what the best way to deal with those method overrides is.

Also, there's still a bunch of errors in `visual/animation`, which I could fix, but it would require some type assertions here and there (i.e. actually raising an error if the type expectation is not met) and I didn't want to go and break @kaihsin animation scripts.

We can discuss & continue here, or just merge (if the changes are fine) and continue in a separate PR, since this already improves things a bit.

### CODE REVIEW

**Comment by kaihsin** [comment]
For the CU3 bug, would you mind also adding unit tests for one of the angle to be zero?

**Comment by david-pl** [comment]
Oh, CI fails because I added a method `get_present_trait` to kirin locally (just hitting F12 in vscode can do that). I'll open another PR there, because I think it makes sense to have it. But I we can also use `cast` here to assert types for the linter.

File: src/bloqade/analysis/address/impls.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def unwrap(
         stmt: squin.wire.Unwrap,
     ):
 
-        origin_qubit = frame.get(stmt.qubit)
+        origin_qubit = frame.get_casted(stmt.qubit, AddressQubit)
 
         return (AddressWire(origin_qubit=origin_qubit),)
```
Comment (weinbe58) [inline src/bloqade/analysis/address/impls.py:195]:
```suggestion
        origin_qubit = frame.get(stmt.qubit)
        
        if isintance(origin_qubit, AddressQubit):

            return (AddressWire(origin_qubit=origin_qubit),)
        else:
            return (Address.top(), )
```

I think the implementation of this function is wrong maybe @johnzl-777 can confirm if this makes sense?

File: src/bloqade/analysis/address/impls.py [hunk 2]
```diff
@@ -203,7 +203,10 @@ def apply(
...
 
         origin_qubits = tuple(
-            [frame.get(input_elem).origin_qubit for input_elem in stmt.inputs]
+            [
+                frame.get_casted(input_elem, AddressWire).origin_qubit
+                for input_elem in stmt.inputs
+            ]
         )
         new_address_wires = tuple(
             [AddressWire(origin_qubit=origin_qubit) for origin_qubit in origin_qubits]
```
Comment (weinbe58) [inline src/bloqade/analysis/address/impls.py:212]: This is also wrong you should just return all the values from the frame because there is no dependency on what the value is. 

```
return frame.get_values(stmt.inputs)
```

**Review by weinbe58 — COMMENTED** [review]
LGTM but I think we might have some issues with the implementation in some areas not just linting.

File: src/bloqade/analysis/address/impls.py (hunk 1 above)
Comment (johnzl-777) [reply src/bloqade/analysis/address/impls.py:195]: Yes, it does! Thank you for catching this (and explaining how to handle these pyright complaints moving forward 👍 )

File: src/bloqade/analysis/address/impls.py (hunk 2 above)
Comment (david-pl) [reply src/bloqade/analysis/address/impls.py:212]: @johnzl-777 how about this one?

Comment (weinbe58) [reply src/bloqade/analysis/address/impls.py:216]: I talked to @johnzl-777 in person, he is OK with this change.

**Comment by Roger-luo** [comment]
fyi. https://github.com/QuEraComputing/kirin/pull/370 is in v0.16.7 now

**Comment by Roger-luo** [comment]
test failure seems due to a simple typo

File: src/bloqade/analysis/address/impls.py (hunk 2 above)
Comment (david-pl) [reply src/bloqade/analysis/address/impls.py:216]: Changed it!

**Comment by Roger-luo** [comment]
bump @weinbe58 for a review.

**Review by weinbe58 — APPROVED** [review]
LGTM
